.github/workflows/ci.yml

```diff
@@ -182,28 +193,44 @@ jobs:
           SIGNING_KEY: ${{ secrets.SIGNING_KEY }}
         run: .github/ci.sh sign $NAME-with-solvers.tar.gz
 
-      - if: matrix.ghc == '8.10.7'
+      - if: matrix.ghc == '8.10.7' && matrix.hpc == false
         uses: actions/upload-artifact@v2
         with:
           name: ${{ steps.config.outputs.name }} (GHC ${{ matrix.ghc }})
           path: "${{ steps.config.outputs.name }}.tar.gz*"
           if-no-files-found: error
           retention-days: ${{ needs.config.outputs.retention-days }}
 
-      - if: matrix.ghc == '8.10.7'
+      - if: matrix.ghc == '8.10.7' && matrix.hpc == false
         uses: actions/upload-artifact@v2
         with:
           name: ${{ steps.config.outputs.name }}-with-solvers (GHC ${{ matrix.ghc }})
           path: "${{ steps.config.outputs.name }}-with-solvers.tar.gz*"
           if-no-files-found: error
           retention-days: ${{ needs.config.outputs.retention-days }}
 
-      - if: matrix.ghc == '8.10.7' && matrix.run-tests
+      - if: matrix.ghc == '8.10.7' && matrix.run-tests && matrix.hpc == false
         uses: actions/upload-artifact@v2
         with:
           path: dist/bin
           name: ${{ runner.os }}-bins
 
+      - if: matrix.run-tests && matrix.hpc == true
```

Comment:
Here (and in the next two steps), why are we concerned with `matrix.run_tests`?

In fact, for this entire job, it's not clear to me what `matrix.run_tests` is intended to solve, and why we need it at all in this job.  Any ideas?

Reply:
It looks like `matrix.run_tests` exists to avoid uploading build artifacts from the `ubuntu-20.04` build because none of the testing runs use that version of Ubuntu. Does that sound right to you?

Reply:
That being said, `run_tests` is implicitly true for the HPC builds so I removed it from the HPC related steps.